File: php/rps.php
<?php

$choices = array(
  // choice => beats
  'paper' => 'rock',
  'rock' => 'scissors',
  'scissors' => 'paper'
);

// play until someome wins
while ( true ) {
  // calculate the computer move first, no cheating
  $computer_move = $choices[array_rand($choices)];

  // get player move, must be valid
  do {
    echo("Rock, Paper, or Scissors? ");

    $human_move = strtolower(trim(fgets( STDIN )));
  } while ( !in_array($human_move, $choices) );

  // calculate winner
  if ( $human_move == $computer_move ) {
    echo("Tie!\n");
  } else if ( $choices[$human_move] == $computer_move ) {
    echo("Human Wins!\n");
    break;
  } else if ( $choices[$computer_move] == $human_move ) {
    echo("Computer Wins!\n");
    break;
  }
}

File: julia/rps.jl

import Random
function rock_paper_scissors()
    options = ["rock", "paper", "scissors"]

    while true
        print("Your Move: ")

        player_turn = readline()

        player_turn = lowercase(player_turn)

        opponent_turn = rand(1:3)
       
        opponent_choice = options[opponent_turn]
        
        if player_turn in options

            if player_turn == opponent_choice
                println("Tie, Replay")
            elseif player_turn == "paper" && opponent_choice == "rock"
                println("You, Win")
                break
            elseif player_turn == "scissors" && opponent_choice =="paper"
                println("You, Win")
                break
            elseif player_turn == "rock" && opponent_choice =="scissors"
                println("You, Win")
                break
            else
                println("Computer, Wins")
                break

            end
        end
    end
end

rock_paper_scissors()